index.js

```diff
@@ -489,6 +496,15 @@ function updateWriteNT () {
   this.update()
 }
 
+function tickDrains (drains) {
```

Comment:
I think it assumes that `writes` is always >=1 so condition is always met, it's just an internal helper, so it can assume things like that

Reply:
I think I understood the code now, and I don't think there's a bug, but the code is not very pretty IMO, although it's efficient, which I think is the goal here.

To clarify: it relies on drains being reverse-ordered by the amount of writes, which I guess is always true. So it's guaranteed that the if-condition only triggers for the first X entries, because If drains[i].writes > 0, then drains[i+1]  > 0

Reply:
writes are monotonic, is, `drains[i].writes < drives[i + 1].writes`